src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyRelayedResponse.java

```diff
@@ -126,43 +126,52 @@ private void onResponseBegin(
         final long affinity = begin.affinity();
         final OctetsFW extension = begin.extension();
 
-        final HttpBeginExFW httpBeginEx = extension.get(factory.httpBeginExRO::wrap);
-        final boolean hasEtag = httpBeginEx.headers().matchFirst(h -> ETAG.equals(h.name().asString())) != null;
-        final HttpHeaderFW status = httpBeginEx.headers().matchFirst(h -> STATUS.equals(h.name().asString()));
-        final int statusGroup = status != null ? parseInt(status.value().asString()) / 100 : 0;
-        final Consumer<ArrayFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW>> headers = hs ->
+        final HttpBeginExFW httpBeginEx = extension.get(factory.httpBeginExRO::tryWrap);
+        if (httpBeginEx != null)
```

Comment:
No, the idea is not to treat `httpBeginEx == null` as a special case, instead just default it and treat it the same as if those headers had arrived with the `BEGIN` frame explicitly.